vendor/bat-native-ads/src/bat/ads/internal/federated/covariate_log_entry.h

```diff
@@ -0,0 +1,29 @@
+/* Copyright 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_VENDOR_BAT_NATIVE_ADS_SRC_BAT_ADS_INTERNAL_FEDERATED_COVARIATE_LOGS_ENTRY_H_
+#define BRAVE_VENDOR_BAT_NATIVE_ADS_SRC_BAT_ADS_INTERNAL_FEDERATED_COVARIATE_LOGS_ENTRY_H_
+
+#include <string>
+
+#include "bat/ads/public/interfaces/ads.mojom.h"
+
+namespace ads {
+
+class CovariateLogEntry {
+ public:
+  CovariateLogEntry();
+  virtual ~CovariateLogEntry();
+
+  virtual mojom::DataType GetDataType() const = 0;
+  virtual mojom::CovariateType GetCovariateType() const = 0;
+  virtual std::string GetValue() const = 0;
+
+  // TODO(Moritz Haller): move ads.mojom to federated.mojom and import
```

Comment:
Are we making this change now, or later? If later can we remove the comment and add a new issue in GitHub

Reply:
added ticket here https://github.com/brave/brave-browser/issues/21188